File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/core/command/FuelPumpSetCmd.kt
package com.syncrhrohealth.bandpass_ble_connection.core.command

import com.syncrhrohealth.bandpass_ble_connection.core.command.builder.CommandBuilder
import com.syncrhrohealth.bandpass_ble_connection.core.enum.CommandCode
import com.syncrhrohealth.bandpass_ble_connection.core.handler.DeviceHandler
import java.nio.ByteBuffer
import java.nio.ByteOrder

/** Data format
 *
 *  float       fuel_pump
 *
 * */

object FuelPumpSetCmd {
    fun send(fuelPump: Float, handler: DeviceHandler) {
        val buffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN)

        buffer.putFloat(fuelPump)

        val data = buffer.array()

        val packet = CommandBuilder.build(
            CommandCode.PACKET_CMD_FUEL_PUMP_SET,
            data
        )

        handler.write(packet)
    }
}

File: ios/Classes/Core/Command/FuelPumpSetCmd.swift
//
//  FuelPumpSetCmd.swift
//  Runner
//
//  Created by MAC on 11/3/25.
//

import Foundation

/// Data format:
/// - `float fuel_pump` (4 bytes, little-endian)
struct FuelPumpSetCmd {
    static func send(fuelPump: Float, handler: DeviceHandler) {
        var data = Data()
        
        // Convert fuelPump (Float) to Data (4 bytes, little-endian)
        let fuelPumpData = Data.Float32ToData(fuelPump, byteOder: .LittleEndian)
        data.append(fuelPumpData)
        
        // Build the packet with CommandBuilder
        let packet = CommandBuilder.build(commandCode: .packetCmdFuelPumpSet, data: data)
        
        // Send packet
        handler.write(data: packet)
    }
}
